### PR TITLE
Use arrow functions for anonymous functions

### PR DESCRIPTION
**Pull request by teoli2003**
When possible, it is better to use arrow functions instead of anonymous functions

### CODE REVIEW

File: files/en-us/web/api/audiobuffersourcenode/loopstart/index.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ function getData() {
         loopendControl.setAttribute('max', Math.floor(songLength));
       },
 
-      function(e){`Error with decoding audio data: ${e.err}`});
+      (e) => {`Error with decoding audio data: ${e.err}`});
```
Comment (Josh-Cena) [inline files/en-us/web/api/audiobuffersourcenode/loopstart/index.md:69]: I dunno... `console.error`?

File: files/en-us/web/api/audiobuffersourcenode/playbackrate/index.md [hunk 2]
```diff
@@ -59,18 +59,18 @@ function getData() {
...
+    audioCtx.decodeAudioData(audioData, (buffer) => {
         myBuffer = buffer;
         source.buffer = myBuffer;
         source.playbackRate.value = playbackControl.value;
         source.connect(audioCtx.destination);
         source.loop = true;
       },
 
-      function(e){`Error with decoding audio data: ${e.err}`});
+      (e) => {`Error with decoding audio data: ${e.err}`});
```
Comment (Josh-Cena) [inline files/en-us/web/api/audiobuffersourcenode/playbackrate/index.md:73]: This should be logged?

**Comment by Josh-Cena** [comment]
GitHub is lagging for me, can't view the rest of the files😥 Will try tomorrow

**Comment by teoli2003** [comment]
> GitHub is lagging for me, can't view the rest of the files😥 Will try tomorrow

Same problem for me.

File: files/en-us/web/api/audiocontext/createmediastreamdestination/index.md [hunk 3]
```diff
@@ -54,38 +54,38 @@ From here, you can play and save the opus file.
...
-     const chunks = [];
-     const ac = new AudioContext();
-     const osc = ac.createOscillator();
-     const dest = ac.createMediaStreamDestination();
-     const mediaRecorder = new MediaRecorder(dest.stream);
-     osc.connect(dest);
-
-     b.addEventListener("click", function(e) {
-       if (!clicked) {
+      const b = document.querySelector("button");
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/audiocontext/createmediastreamdestination/index.md:57]: Yes, but this should be in a follow-up PR. (Else it is a bit of rabbit hole)

File: files/en-us/web/api/audiobuffersourcenode/loop/index.md [hunk 4]
```diff
@@ -52,18 +52,18 @@ function getData() {
...
+    audioCtx.decodeAudioData(audioData, (buffer) => {
         myBuffer = buffer;
         source.buffer = myBuffer;
         source.playbackRate.value = playbackControl.value;
         source.connect(audioCtx.destination);
         source.loop = true;
       },
 
-      function(e){`Error with decoding audio data: ${e.err}`});
+      (e) => {`Error with decoding audio data: ${e.err}`});
```
Comment (Josh-Cena) [inline files/en-us/web/api/audiobuffersourcenode/loop/index.md:66]:
```suggestion
      (e) => console.error(`Error with decoding audio data: ${e.err}`));
```

File: files/en-us/web/api/audiobuffersourcenode/loopend/index.md [hunk 5]
```diff
@@ -74,7 +74,7 @@ function getData() {
         loopendControl.setAttribute('max', Math.floor(songLength));
       },
 
-      function(e){`Error with decoding audio data: ${e.err}`});
+      (e) => {`Error with decoding audio data: ${e.err}`});
```
Comment (Josh-Cena) [inline files/en-us/web/api/audiobuffersourcenode/loopend/index.md:77]:
```suggestion
      (e) => console.error(`Error with decoding audio data: ${e.err}`));
```

File: files/en-us/web/api/audiobuffersourcenode/loopstart/index.md [hunk 6]
```diff
@@ -66,7 +66,7 @@ function getData() {
         loopendControl.setAttribute('max', Math.floor(songLength));
       },
 
-      function(e){`Error with decoding audio data: ${e.err}`});
+      (e) => {console.error(`Error with decoding audio data: ${e.err}`)});
```
Comment (Josh-Cena) [inline files/en-us/web/api/audiobuffersourcenode/loopstart/index.md:69]:
```suggestion
      (e) => console.error(`Error with decoding audio data: ${e.err}`));
```

File: files/en-us/web/api/audiobuffersourcenode/playbackrate/index.md [hunk 7]
```diff
@@ -59,18 +59,18 @@ function getData() {
...
+    audioCtx.decodeAudioData(audioData, (buffer) => {
         myBuffer = buffer;
         source.buffer = myBuffer;
         source.playbackRate.value = playbackControl.value;
         source.connect(audioCtx.destination);
         source.loop = true;
       },
 
-      function(e){`Error with decoding audio data: ${e.err}`});
+      (e) => {console.error(`Error with decoding audio data: ${e.err}`)});
```
Comment (Josh-Cena) [inline files/en-us/web/api/audiobuffersourcenode/playbackrate/index.md:73]:
```suggestion
      (e) => console.error(`Error with decoding audio data: ${e.err}`));
```

File: files/en-us/web/api/audiocontext/createmediastreamdestination/index.md [hunk 8]
```diff
@@ -54,38 +54,38 @@ From here, you can play and save the opus file.
...
+      });
+
+      mediaRecorder.ondataavailable = (evt) => {
+        // Push each chunk (blobs) in an array
+        chunks.push(evt.data);
+      };
+
+      mediaRecorder.onstop = (evt) => {
+        // Make blob out of our blobs, and open it.
+        const blob = new Blob(chunks, { 'type' : 'audio/ogg; codecs=opus' });
```
Comment (Josh-Cena) [inline files/en-us/web/api/audiocontext/createmediastreamdestination/index.md:86]:
```suggestion
        const blob = new Blob(chunks, { type: 'audio/ogg; codecs=opus' });
```

File: files/en-us/web/api/audiocontext/resume/index.md [hunk 9]
```diff
@@ -45,13 +45,13 @@ The following snippet is taken from our [AudioContext states demo](https://githu
 the button is updated as appropriate once the promise resolves.
 
 ```js
-susresBtn.onclick = function() {
+susresBtn.onclick = () => {
   if(audioCtx.state === 'running') {
```
Comment (Josh-Cena) [inline files/en-us/web/api/audiocontext/resume/index.md:49]:
```suggestion
  if (audioCtx.state === 'running') {
```

File: files/en-us/web/api/audiocontext/resume/index.md [hunk 10]
```diff
@@ -45,13 +45,13 @@ The following snippet is taken from our [AudioContext states demo](https://githu
...
-    audioCtx.suspend().then(function() {
+    audioCtx.suspend().then(() => {
       susresBtn.textContent = 'Resume context';
     });
   } else if(audioCtx.state === 'suspended') {
```
Comment (Josh-Cena) [inline files/en-us/web/api/audiocontext/resume/index.md:53]:
```suggestion
  } else if (audioCtx.state === 'suspended') {
```

File: files/en-us/web/api/audiocontext/suspend/index.md [hunk 11]
```diff
@@ -37,13 +37,13 @@ A {{jsxref("Promise")}} that resolves with {{jsxref('undefined')}}. The promise
 The following snippet is taken from our [AudioContext states demo](https://github.com/mdn/webaudio-examples/blob/master/audiocontext-states/index.html) ([see it running live](https://mdn.github.io/webaudio-examples/audiocontext-states/).) When the suspend/resume button is clicked, the {{domxref("BaseAudioContext/state", "AudioContext.state")}} is queried — if it is `running`, `suspend()` is called; if it is `suspended`, {{domxref("AudioContext/resume", "resume()")}} is called. In each case, the text label of the button is updated as appropriate once the promise resolves.
 
 ```js
-susresBtn.onclick = function() {
+susresBtn.onclick = () => {
   if(audioCtx.state === 'running') {
```
Comment (Josh-Cena) [inline files/en-us/web/api/audiocontext/suspend/index.md:41]:
```suggestion
  if (audioCtx.state === 'running') {
```

File: files/en-us/web/api/audiocontext/suspend/index.md [hunk 12]
```diff
@@ -37,13 +37,13 @@ A {{jsxref("Promise")}} that resolves with {{jsxref('undefined')}}. The promise
...
-    audioCtx.suspend().then(function() {
+    audioCtx.suspend().then(() => {
       susresBtn.textContent = 'Resume context';
     });
   } else if(audioCtx.state === 'suspended') {
```
Comment (Josh-Cena) [inline files/en-us/web/api/audiocontext/suspend/index.md:45]:
```suggestion
  } else if (audioCtx.state === 'suspended') {
```

File: files/en-us/web/api/authenticatorassertionresponse/authenticatordata/index.md [hunk 13]
```diff
@@ -46,14 +46,14 @@ const options = {
...
     const authenticatorData = assertionPKCred.response.authenticatorData;
     // Maybe try to convert the authenticatorData to see what's inside
 
     // Send response and client extensions to the server so that it can
     // go on with the authentication
 
-}).catch(function (err) {
+}).catch((err) => {
    console.error(err);
 });
```
Comment (Josh-Cena) [inline files/en-us/web/api/authenticatorassertionresponse/authenticatordata/index.md:58]:
```suggestion
  })
  .catch((err) => console.error(err));
```

File: files/en-us/web/api/authenticatorassertionresponse/index.md [hunk 14]
```diff
@@ -45,13 +45,13 @@ const options = {
...
+    .then((credentialInfoAssertion) => {
     const assertionResponse = credentialInfoAssertion.response;
     // Do something specific with the response
 
     // send assertion response back to the server
     // to proceed with the control of the credential
-}).catch(function (err) {
+}).catch((err) => {
      console.error(err);
 });
```
Comment (Josh-Cena) [inline files/en-us/web/api/authenticatorassertionresponse/index.md:56]:
```suggestion
  })
  .catch((err) => console.error(err));
```

File: files/en-us/web/api/authenticatorassertionresponse/signature/index.md [hunk 15]
```diff
@@ -48,13 +48,13 @@ const options = {
...
+  .then((assertionPKCred) => {
     const signature = assertionPKCred.response.signature;
 
     // Send response and client extensions to the server so that it can
     // go on with the authentication
 
-}).catch(function (err) {
+}).catch((err) => {
    console.error(err);
 });
```
Comment (Josh-Cena) [inline files/en-us/web/api/authenticatorassertionresponse/signature/index.md:59]:
```suggestion
  })
  .catch((err) => console.error(err));
```

File: files/en-us/web/api/authenticatorassertionresponse/userhandle/index.md [hunk 16]
```diff
@@ -45,13 +45,13 @@ const options = {
...
+  .then((assertionPKCred) => {
     const userHandle = assertionPKCred.response.userHandle;
 
     // Send response and client extensions to the server so that it can
     // go on with the authentication
 
-}).catch(function (err) {
+}).catch((err) => {
    console.error(err);
 });
```
Comment (Josh-Cena) [inline files/en-us/web/api/authenticatorassertionresponse/userhandle/index.md:56]:
```suggestion
  })
  .catch((err) => console.error(err));
```

File: files/en-us/web/api/authenticatorattestationresponse/attestationobject/index.md [hunk 17]
```diff
@@ -86,13 +86,13 @@ const publicKey = {
...
+  .then((newCredentialInfo) => {
     const attestationObj = newCredentialInfo.response.attestationObject;
     // This will be a CBOR encoded ArrayBuffer
 
     // Do something with the response
     // (sending it back to the relying party server maybe?)
-  }).catch(function (err) {
+  }).catch((err) => {
      console.error(err);
   });
```
Comment (Josh-Cena) [inline files/en-us/web/api/authenticatorattestationresponse/attestationobject/index.md:97]:
```suggestion
  })
  .catch((err) => console.error(err));
```

File: files/en-us/web/api/authenticatorattestationresponse/gettransports/index.md [hunk 18]
```diff
@@ -73,10 +73,10 @@ const publicKey = {
...
 
 navigator.credentials.create({ publicKey })
-  .then(function (newCredentialInfo) {
+  .then((newCredentialInfo) => {
     const transports = newCredentialInfo.response.getTransports();
     console.table(transports); // may be something like ["internal", "nfc", "usb"]
-  }).catch(function (err) {
+  }).catch((err) => {
      console.error(err);
   });
```
Comment (Josh-Cena) [inline files/en-us/web/api/authenticatorattestationresponse/gettransports/index.md:81]:
```suggestion
  })
  .catch((err) => console.error(err));
```

File: files/en-us/web/api/authenticatorattestationresponse/index.md [hunk 19]
```diff
@@ -57,11 +57,11 @@ const publicKey = {
...
 navigator.credentials.create({ publicKey })
-  .then(function (newCredentialInfo) {
+  .then((newCredentialInfo) => {
     const response = newCredentialInfo.response;
     // Do something with the response
     // (sending it back to the relying party server maybe?)
-  }).catch(function (err) {
+  }).catch((err) => {
      console.error(err);
   });
```
Comment (Josh-Cena) [inline files/en-us/web/api/authenticatorattestationresponse/index.md:66]:
```suggestion
  })
  .catch((err) => console.error(err));
```

File: files/en-us/web/api/authenticatorresponse/index.md [hunk 20]
```diff
@@ -42,11 +42,11 @@ const options = {
...
 navigator.credentials.get({ "publicKey": options })
-    .then(function (credentialInfoAssertion) {
+    .then((credentialInfoAssertion) => {
     const assertionResponse = credentialInfoAssertion.response;
     // send assertion response back to the server
     // to proceed with the control of the credential
-}).catch(function (err) {
+}).catch((err) => {
      console.error(err);
 });
```
Comment (Josh-Cena) [inline files/en-us/web/api/authenticatorresponse/index.md:51]:
```js
navigator.credentials.get({ publicKey: options })
  .then((credentialInfoAssertion) => {
    const assertionResponse = credentialInfoAssertion.response;
    // send assertion response back to the server
    // to proceed with the control of the credential
  })
  .catch((err) => console.error(err));
```

File: files/en-us/web/api/authenticatorresponse/index.md [hunk 21]
```diff
@@ -74,9 +74,9 @@ const publicKey = {
 };
 
 navigator.credentials.create({ publicKey })
-  .then(function (newCredentialInfo) {
+  .then((newCredentialInfo) => {
     const attestationResponse = newCredentialInfo.response;
-  }).catch(function (err) {
+  }).catch((err) => {
      console.error(err);
   });
```
Comment (Josh-Cena) [inline files/en-us/web/api/authenticatorresponse/index.md:81]:
```suggestion
  })
  .catch((err) => console.error(err));
```

File: files/en-us/web/api/baseaudiocontext/createconvolver/index.md [hunk 22]
```diff
@@ -60,13 +60,13 @@ ajaxRequest = new XMLHttpRequest();
...
-ajaxRequest.onload = function() {
+ajaxRequest.onload = () => {
   const audioData = ajaxRequest.response;
-  audioCtx.decodeAudioData(audioData, function(buffer) {
+  audioCtx.decodeAudioData(audioData, (buffer) => {
       concertHallBuffer = buffer;
       soundSource = audioCtx.createBufferSource();
       soundSource.buffer = concertHallBuffer;
-    }, function(e){`Error with decoding audio data: ${e.err}`});
+    }, (e) => {`Error with decoding audio data: ${e.err}`});
```
Comment (Josh-Cena) [inline files/en-us/web/api/baseaudiocontext/createconvolver/index.md:69]:
```suggestion
    }, (e) => console.error(`Error with decoding audio data: ${e.err}`));
```

File: files/en-us/web/api/baseaudiocontext/createscriptprocessor/index.md [hunk 23]
```diff
@@ -92,20 +92,20 @@ function getData() {
...
+  request.onload = () => {
     const audioData = request.response;
 
-    audioCtx.decodeAudioData(audioData, function(buffer) {
+    audioCtx.decodeAudioData(audioData, (buffer) => {
       myBuffer = buffer;
       source.buffer = myBuffer;
     },
-    function(e){`Error with decoding audio data: ${e.err}`});
+    (e) => {`Error with decoding audio data: ${e.err}`});
```
Comment (Josh-Cena) [inline files/en-us/web/api/baseaudiocontext/createscriptprocessor/index.md:102]:
```suggestion
    (e) => console.error(`Error with decoding audio data: ${e.err}`));
```

File: files/en-us/web/api/baseaudiocontext/decodeaudiodata/index.md [hunk 24]
```diff
@@ -103,17 +103,17 @@ function getData() {
...
-    audioCtx.decodeAudioData(audioData, function(buffer) {
+    audioCtx.decodeAudioData(audioData, (buffer) => {
         source.buffer = buffer;
 
         source.connect(audioCtx.destination);
         source.loop = true;
       },
 
-      function(e){ console.log(`Error with decoding audio data: ${e.err}`); });
+      (err) => { console.error(`Error with decoding audio data: ${err.err}`); });
```
Comment (Josh-Cena) [inline files/en-us/web/api/baseaudiocontext/decodeaudiodata/index.md:116]:
```suggestion
      (err) => console.error(`Error with decoding audio data: ${err.err}`));
```

File: files/en-us/web/api/bluetooth/requestdevice/index.md [hunk 25]
```diff
@@ -81,11 +81,11 @@ let options = {
...
 
-navigator.bluetooth.requestDevice(options).then(function(device) {
+navigator.bluetooth.requestDevice(options).then((device) => {
   console.log(`Name: ${device.name}`);
   // Do something with the device.
 })
-.catch(function(error) {
+.catch((error) => {
   console.log(`Something went wrong. ${error}`);
 });
```
Comment (Josh-Cena) [inline files/en-us/web/api/bluetooth/requestdevice/index.md:90]:
```suggestion
.catch((error) => console.error(`Something went wrong. ${error}`));
```

File: files/en-us/web/api/cache/add/index.md [hunk 26]
```diff
@@ -59,9 +59,9 @@ A {{jsxref("Promise")}} that resolves with `undefined`.
 This code block waits for an {{domxref("InstallEvent")}} to fire, then calls {{domxref("ExtendableEvent.waitUntil","waitUntil()")}} to handle the install process for the app. This consists of calling {{domxref("CacheStorage.open")}} to create a new cache, then using {{domxref("Cache.add")}} to add an asset to it.
 
 ```js
-this.addEventListener('install', function(event) {
+this.addEventListener('install', (event) => {
   event.waitUntil(
-    caches.open('v1').then(function(cache) {
+    caches.open('v1').then((cache) => {
       return cache.add('/sw-test/index.html');
     })
```
Comment (Josh-Cena) [inline files/en-us/web/api/cache/add/index.md:66]:
```suggestion
    caches.open('v1').then((cache) => cache.add('/sw-test/index.html'))
```

File: files/en-us/web/api/cache/match/index.md [hunk 27]
```diff
@@ -80,15 +80,15 @@ response with an response code in the 4xx or 5xx range, the `catch()` will
...
       event.request.headers.get('accept').includes('text/html')) {
     console.log('Handling fetch event for', event.request.url);
     event.respondWith(
-      fetch(event.request).catch(function(e) {
+      fetch(event.request).catch((e) => {
         console.error('Fetch failed; returning offline page instead.', e);
-        return caches.open(OFFLINE_CACHE).then(function(cache) {
+        return caches.open(OFFLINE_CACHE).then((cache) => {
           return cache.match(OFFLINE_URL);
         });
```
Comment (Josh-Cena) [inline files/en-us/web/api/cache/match/index.md:93]:
```suggestion
        return caches.open(OFFLINE_CACHE).then((cache) => cache.match(OFFLINE_URL));
```

File: files/en-us/web/api/cache/put/index.md [hunk 28]
```diff
@@ -82,15 +82,15 @@ like so:
...
   response = r;
-  caches.open('v1').then(function(cache) {
+  caches.open('v1').then((cache) => {
     cache.put(event.request, response);
   });
   return response.clone();
-}).catch(function() {
+}).catch(() => {
   return caches.match('/sw-test/gallery/myLittleVader.jpg');
 });
```
Comment (Josh-Cena) [inline files/en-us/web/api/cache/put/index.md:95]:
```suggestion
}).catch(() => caches.match('/sw-test/gallery/myLittleVader.jpg'));
```

File: files/en-us/web/api/cachestorage/has/index.md [hunk 29]
```diff
@@ -42,15 +42,15 @@ The following example first checks whether a cache called 'v1' exists. If so, we
...
 ```js
-caches.has('v1').then(function(hasCache) {
+caches.has('v1').then((hasCache) => {
   if (!hasCache) {
     someCacheSetupFunction();
   } else {
-    caches.open('v1').then(function(cache) {
+    caches.open('v1').then((cache) => {
       return cache.addAll(myAssets);
     });
```
Comment (Josh-Cena) [inline files/en-us/web/api/cachestorage/has/index.md:51]:
```suggestion
    caches.open('v1').then((cache) => cache.addAll(myAssets));
```

File: files/en-us/web/api/cachestorage/index.md [hunk 30]
```diff
@@ -79,24 +79,24 @@ self.addEventListener('install', function(event) {
...
 
-        caches.open('v1').then(function (cache) {
+        caches.open('v1').then((cache) => {
           cache.put(event.request, responseClone);
         });
         return response;
-      }).catch(function () {
+      }).catch(() => {
         return caches.match('/sw-test/gallery/myLittleVader.jpg');
       });
```
Comment (Josh-Cena) [inline files/en-us/web/api/cachestorage/index.md:101]:
```suggestion
      }).catch(() => caches.match('/sw-test/gallery/myLittleVader.jpg'));
```

File: files/en-us/web/api/cachestorage/match/index.md [hunk 31]
```diff
@@ -85,24 +85,24 @@ like so:
...
 
-        caches.open('v1').then(function (cache) {
+        caches.open('v1').then((cache) => {
           cache.put(event.request, responseClone);
         });
         return response;
-      }).catch(function () {
+      }).catch(() => {
         return caches.match('/sw-test/gallery/myLittleVader.jpg');
       });
```
Comment (Josh-Cena) [inline files/en-us/web/api/cachestorage/match/index.md:107]:
```suggestion
      }).catch(() => caches.match('/sw-test/gallery/myLittleVader.jpg'));
```

File: files/en-us/web/api/convolvernode/normalize/index.md [hunk 32]
```diff
@@ -45,13 +45,13 @@ ajaxRequest = new XMLHttpRequest();
...
-ajaxRequest.onload = function() {
+ajaxRequest.onload = () => {
   let audioData = ajaxRequest.response;
-  audioCtx.decodeAudioData(audioData, function(buffer) {
+  audioCtx.decodeAudioData(audioData, (buffer) => {
       concertHallBuffer = buffer;
       soundSource = audioCtx.createBufferSource();
       soundSource.buffer = concertHallBuffer;
-    }, function(e){`Error with decoding audio data${e.err}`});
+    }, (e) => {`Error with decoding audio data${e.err}`});
```
Comment (Josh-Cena) [inline files/en-us/web/api/convolvernode/normalize/index.md:54]:
```suggestion
    }, (e) => console.error(`Error with decoding audio data: ${e.err}`));
```

File: files/en-us/web/api/convolvernode/buffer/index.md [hunk 33]
```diff
@@ -40,13 +40,13 @@ ajaxRequest = new XMLHttpRequest();
...
-ajaxRequest.onload = function() {
+ajaxRequest.onload = () => {
   const audioData = ajaxRequest.response;
-  audioCtx.decodeAudioData(audioData, function(buffer) {
+  audioCtx.decodeAudioData(audioData, (buffer) => {
       concertHallBuffer = buffer;
       soundSource = audioCtx.createBufferSource();
       soundSource.buffer = concertHallBuffer;
-    }, function(e){`Error with decoding audio data${e.err}`});
+    }, (e) => {`Error with decoding audio data${e.err}`});
```
Comment (Josh-Cena) [inline files/en-us/web/api/convolvernode/buffer/index.md:49]:
```suggestion
    }, (e) => console.error(`Error with decoding audio data: ${e.err}`));
```

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
Fixing all this code is really disheartening...

**Comment by teoli2003** [comment]
> Fixing all this code is really disheartening...

Yes, Modernizing JS is a significant project. At least it usually doesn't regress and once it is done, it is done.

**Review by Josh-Cena — APPROVED** [review]
I think this is good now. Let's go!

**Pull request merged**